components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/AccessTokenIssuer.java

```diff
@@ -324,6 +324,10 @@ public OAuth2AccessTokenRespDTO issue(OAuth2AccessTokenReqDTO tokenReqDTO)
                     tokReqMsgCtx.getAuthorizedUser() + " and scopes: " + tokenRespDTO.getAuthorizedScopes());
         }
 
+        if (tokenReqDTO.getGrantType().equals(GrantType.AUTHORIZATION_CODE.toString())) {
```

Comment:
Cant we use 
if (grantType.equals(GrantType.AUTHORIZATION_CODE.toString())) {

Reply:
+1,  there is another place to fix from old code. "log.debug("Client Id: " + tokenReqDTO.getClientId() + " is not authorized to use grant type: " +
                        tokenReqDTO.getGrantType());"